tensorflow/python/ops/array_ops.py

```diff
@@ -1085,6 +1085,7 @@ def boolean_mask(tensor, mask, name="boolean_mask"):
     tensor:  N-D tensor.
     mask:  K-D boolean tensor, K <= N and K must be known statically.
     name:  A name for this operation (optional).
+    axis: The axis in `tensor` to mask from. Defaults to the first dimension.
```

Comment:
So numpy.compress only allows 1-D boolean masks, which makes the choice of only using an integer for the axis straightforward.  This op allows K-D boolean masks, which raises the question of whether the axis should be (allowed to be) a length-K vector.